src/pages/HistoryPage/EpochForm.tsx

```diff
@@ -422,121 +682,242 @@ const EpochForm = ({
-                          }}
-                        >
-                          <FormTimePicker
-                            onBlur={onBlur}
-                            onChange={onChange}
-                            value={value}
-                            disabled={
-                              selectedEpoch &&
-                              currentEpoch?.id === selectedEpoch?.id
-                            }
-                          />
-                        </Box>
+                      name="repeat"
+                      defaultValue="custom"
+                      render={({ field: { onChange, value } }) => (
+                        <Select
+                          css={{ minWidth: '280px' }}
+                          options={repeat}
+                          value={value}
+                          disabled={shouldFormBeDisabled}
+                          onValueChange={onChange}
+                          id="repeat_type"
+                          label="Cycles"
+                        />
                       )}
                     />
-                    <Text size="medium">
-                      In your
-                      <br /> local timezone
-                    </Text>
+                  </Flex>
+                  <Flex
+                    column
+                    css={{
+                      alignItems: 'flex-start',
+                      maxWidth: '280px',
+                      gap: '$xs',
+                      display:
+                        getValues('repeat') === 'custom' ? 'flex' : 'none',
+                    }}
+                  >
+                    <FormLabel type="label" css={{ fontWeight: '$bold' }}>
+                      Start On{' '}
+                      <Tooltip content="The first day of the epoch in your local time zone">
+                        <Info size="sm" />
+                      </Tooltip>
+                    </FormLabel>
+                    <FormDatePicker
+                      disabled={shouldFormBeDisabled}
+                      control={control}
+                      id="custom_start_date"
+                      name="custom_start_date"
+                      css={{ minWidth: '280px' }}
+                    />
+                    <Flex row css={{ gap: '$sm' }}>
+                      <Box
+                        css={{
+                          maxWidth: '80px',
+                          '> div': { mb: '0 !important' },
+                        }}
+                      >
+                        <FormInputField
+                          number
+                          label="Length"
+                          infoTooltip="The duration of each epoch"
+                          disabled={shouldFormBeDisabled}
+                          defaultValue={1}
+                          inputProps={{ min: 1 }}
+                          control={control}
+                          id="custom_duration_qty"
```

Comment:
I was able to make an epoch that was 100 months and repeats every 1000. Do we want any bounds?

Reply:
Also if you just keep typing 0s in there it crashes the app, eg 1000000000000 you get a NaN

Reply:
these could prob just be capped at 3 chars